### PR TITLE
Send metrics to riemann.

### PR DESCRIPTION
**Pull request by anthonygalea**

### CODE REVIEW

File: src/yetibot/core/monitoring.clj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+(s/def ::riemann (s/keys :req-un [::host ::port]))
+
+(s/def ::config (s/keys :req-un [::riemann]))
+
+(defonce riemann-reporter (atom nil))
+
+(defn start
+  []
+  (log/info "starting monitoring")
```
Comment (devth) [inline src/yetibot/core/monitoring.clj]: Would be useful if we only log this when config is valid, and if not maybe a debug statement saying monitoring is not configured.

**Comment by devth** [comment]
Wonder if we could provide some docs or at least links on setting up a monitoring instance. Maybe a docker command or include it in the local `docker-compose.yml`. 🤔

File: src/yetibot/core/monitoring.clj (hunk 1 above)
Comment (anthonygalea) [reply src/yetibot/core/monitoring.clj]: Done. Also made the port optional and interval configurable. What do you think?

**Comment by anthonygalea** [comment]
> Wonder if we could provide some docs or at least links on setting up a monitoring instance. Maybe a docker command or include it in the local `docker-compose.yml`. 🤔

In order to test things locally I had expanded the existing `docker-compose.yml`. See https://github.com/yetibot/yetibot/pull/962. `riemann.config` is a minimal configuration I used to get things going. Besides this I think we could also provide a default grafana dashboard?

File: src/yetibot/core/monitoring.clj (hunk 1 above)
Comment (devth) [reply src/yetibot/core/monitoring.clj]: Perfect!

**Comment by devth** [comment]
Sounds great. Looking forward to testing this out.

**Pull request merged**